treeherder/changelog/collector.py

```diff
@@ -0,0 +1,81 @@
+""" Collector, grabs changes in various sources and put them in a DB.
+"""
+import json
+import os
+
+from treeherder.changelog.filters import Filters
+from treeherder.utils import github
+
+MAX_ITEMS = 100
+CFG = os.path.join(os.path.dirname(__file__), "repositories.json")
+
+with open(CFG) as f:
+    CFG = json.loads(f.read())
+
+
+class GitHub:
+    def __init__(self):
+        self.filters = Filters()
+
+    def get_changes(self, **kw):
+        owner = kw["user"]
+        repository = kw["repository"]
+        filters = kw.get("filters")
+        gh_options = {"number": kw.get("number", MAX_ITEMS)}
+
+        for release in github.get_releases(owner, repository, params=gh_options):
```

Comment:
It looks like the GITHUB_CLIENT_ID/SECRET credentials are missing here per https://github.com/mozilla/treeherder/blob/master/treeherder/etl/push_loader.py#L66

The GITHUB_TOKEN added to the headers in `make_request` is currently for local development use. We'll move away from the key/secret pair and adopt the token for production use later on. So for now, you need the key/secret pair to avoid the rate limiting.